### ISSUE TITLE
hv.operation.image_overlay

### ISSUE THREAD

**Issue by vascotenner** [ioam/holoviews#204]
How to use hv.operation.image_overlay?

I tried: 
```python
hv.operation.image_overlay(Image(data))

---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-108-c978ab22c179> in <module>()
----> 1 hv.operation.image_overlay(Image(nd.filters.gaussian_filter(pump, 2))#, bounds=bounds, kdims=kdims, label='Pump'),
      2                             )#hv.Image(data[6,0], label='{}mW'.format(powers[6]), kdims=kdims, bounds=bounds, group='data'))

/home/tenner/src/param/param/parameterized.pyc in __new__(class_, *args, **params)
   1845         inst = class_.instance()
   1846         inst._set_name(class_.__name__)
-> 1847         return inst.__call__(*args,**params)
   1848 
   1849     def __call__(self,*args,**kw):

/home/tenner/src/holoviews/holoviews/core/operation.pyc in __call__(self, element, **params)
    109 
    110         if isinstance(element, ViewableElement):
--> 111             processed = self._process(element)
    112         elif isinstance(element, GridSpace):
    113             # Initialize an empty axis layout

/home/tenner/src/holoviews/holoviews/operation/element.pyc in _process(self, raster, key)
    236     def _process(self, raster, key=None):
    237         specs = tuple(el.strip() for el in self.p.spec.split('*'))
--> 238         ordering, strengths = self._match_overlay(raster, specs)
    239         if all(el is None for el in ordering):
    240             raise Exception("The image_overlay operation requires at least one match")

/home/tenner/src/holoviews/holoviews/operation/element.pyc in _match_overlay(self, raster, overlay_spec)
    225         for el in elements:
    226             for pos in range(len(overlay_spec)):
--> 227                 strength = self._match(el, overlay_spec[pos])
    228                 if strength is None:               continue  # No match
    229                 elif (strength <= strengths[pos]): continue  # Weaker match

/home/tenner/src/holoviews/holoviews/operation/element.pyc in _match(cls, el, spec)
    201         spec_dict = dict(zip(['type', 'group', 'label'], spec.split('.')))
    202         if not isinstance(el, Image) or spec_dict['type'] != 'Image':
--> 203             raise NotImplementedError("Only Image currently supported")
    204 
    205         strength = 1

NotImplementedError: Only Image currently supported
```

**Comment by jlstevens** [comment]
We are hoping to document operations right after we finally have the new transforming data tutorial uploaded. This particular operation is probably the most tricky one we've included with HoloViews!

Anyway, looking at your code, I think you probably don't need to use an operation for this. If you want to make an ``Overlay`` containing a single image you can use:

```python
hv.Overlay([Image(data)])
```

I'm not entirely sure why you need an ``Overlay`` containing with a single element but this is what I assume you are looking for!

**Comment by vascotenner** [comment]
I wanted to create a RGB image from 2 images. The docstring of operation.image_overlay mention that this is possible

**Comment by jlstevens** [comment]
Ok, now it makes sense - there was only one image in your example!

The ``image_overlay`` operation can be used as long as you have consistent labels that always map to the 'R', 'G' and 'B' channels in a consistent way. If you don't want to use the operation, and you only need this once you can pass an overlay straight to ``RGB`` with something like this (where the blue channel is all zeros):

```python
vdim = hv.Dimension('Channel', range=(0,1) # If channels are normalized!
overlay = Image(R, vdims=[vdim]) * Image(G, vdims=[vdim]) * Image(np.zeros(shape),  vdims=[vdim])
hv.RGB(overlay)
```

If this code gets repeated (with different channels being constant according to group/label), then ``hv.operation.image_overlay`` is indeed suitable. In this case, please tell me what group/label strings you are using and how you want to map them to the RGB channels so I can give you a useful  concrete example...

**Comment by vascotenner** [comment]
```
a = Image(pump, label='pump')
b = Image(data, label='120mW')
```

**Comment by jlstevens** [comment]
I had to make a fix (38ed2b6) but this now works:

```python
import numpy as np
import holoviews as hv

# Specifies how you combine your channels
combine = hv.operation.image_overlay.instance(
        spec='Image.Image.Pump * Image.Image.A_120mw * Image.Image.B')

vdim = hv.Dimension('Channel', range=(0,1)) # If channels are normalized!
two_channels = (hv.Image(np.random.rand(5,5),  vdims=[vdim], label='pump')
              * hv.Image(np.random.rand(5,5), vdims=[vdim], label='120mw'))

rgb = hv.RGB(combine(two_channels))
```

Now you can use this to see how it was combined:

```python
rgb[:,:,'R'] + rgb[:,:,'G'] + rgb[:,:,'B']
```

**Comment by jlstevens** [comment]
I think I will close this issue now that I've made the necessary fix and confirmed the operation is working as expected.
